src/macros/macrorecorder.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <gtest/gtest_prod.h>
+
+#include <QDebug>
+#include <QObject>
+#include <QtCore>
+
+#include "control/controlpushbutton.h"
+#include "engine/channelhandle.h"
+#include "macros/macro.h"
+#include "recording/recordingmanagerbase.h"
+
+// TODO(xerus) add enum for recording_status
+
+enum MacroRecordingState : uint8_t {
+    /// Nothing is going on
+    Disabled,
+    /// Recording is active, but nothing currently going on
+    Armed,
+    /// An Action is currently being recorded
+    Recording,
+};
```

Comment:
1) That is right.
2) The meaning of the states has changed, the documentation you just suggested is the previous one. See https://github.com/mixxxdj/mixxx/pull/2873/commits/3a275f91b6715a0b5cdbc5627f46481a08e97be2
I will put more details on that out, but I was sort of unsure which the best format was, since a graphical illustration of the states could be handy.

Reply:
If you use a lock-free queue you don't need to track this additional state and can revert the states back to the way they were. That would make to code a lot easier I think.